server/webhook_parser.go

```diff
@@ -316,6 +318,77 @@ func quoteIssueComment(comment string) string {
 	return "> " + strings.ReplaceAll(comment, "\n", "\n> ")
 }
 
+func preProcessText(comment string) string {
+	asteriskRegex := regexp.MustCompile(`\*(\w+)\*`)
+	hyphenRegex := regexp.MustCompile(`-(\w+)-`)
+	headingRegex := regexp.MustCompile(`(?m)^(h[1-6]\.)\s+`)
+	codeBlockRegex := regexp.MustCompile(`\{code:[^}]+\}(.+?)\{code\}`)
+	numberedListRegex := regexp.MustCompile(`^#\s+`)
+	colouredTextRegex := regexp.MustCompile(`\{color:[^}]+\}(.*?)\{color\}`)
+	linkRegex := regexp.MustCompile(`\[(.*?)\|([^|\]]+)(?:\|([^|\]]+))?\]`)
+
+	// below code converts lines starting with "#" into a numbered list. It increments the counter if consecutive lines are numbered,
+	// otherwise resets it to 1. The "#" is replaced with the corresponding number and period. Non-numbered lines are added unchanged.
```

Comment:
```suggestion
	// The below code converts lines starting with "#" into a numbered list. It increments the counter if consecutive lines are numbered,
	// otherwise resets it to 1. The "#" is replaced with the corresponding number and period. Non-numbered lines are added unchanged.
```